src/test/java/org/mule/extension/file/integration/FileWriteTestCase.java

```diff
@@ -102,6 +108,33 @@ public void createNewOnNotExistingParentWithoutCreateFolder() throws Exception {
     doWriteOnNotExistingParentWithoutCreateFolder(CREATE_NEW);
   }
 
+  @Test
+  public void writeWithLock() throws Exception {
+    String path = format("%s/%s", temporaryFolder.newFolder().getPath(), TEST_FILENAME);
+    flowRunner("writeWithLock").withVariable("mode", CREATE_NEW)
+        .withVariable("createParent", false)
+        .withVariable("path", path)
+        .withPayload("Hello World!")
+        .run();
+    String content = readPathAsString(path);
+    assertThat(content, is(HELLO_WORLD));
+  }
+
+  @Test
+  public void writeOnLockedFile() throws Exception {
+    File file = temporaryFolder.newFile();
```

Comment:
I think there is a Rule to expect a certain error in the test, using it would make the test more readable.